subsys/bluetooth/host/adv.c

```diff
@@ -747,27 +747,24 @@ static int le_adv_update(struct bt_le_ext_adv *adv,
 			 const struct bt_data *sd, size_t sd_len,
 			 bool ext_adv, bool scannable)
 {
-	struct bt_ad d[2] = {};
-	size_t d_len;
 	int err;
+	struct bt_ad wrapper;
 
 	if (!(ext_adv && scannable)) {
-		d_len = 1;
-		d[0].data = ad;
-		d[0].len = ad_len;
+		wrapper.data = ad;
+		wrapper.len  = ad_len;
 
-		err = set_ad(adv, d, d_len);
+		err = set_ad(adv, &wrapper, 1);
```

Comment:
I would actually have done something like

```suggestion
		err = set_ad(adv, (struct bt_ad){.data = ad, .len = ad_len}, 1);
```

eliminating the need for the variable, or

```suggestion
		const struct bt_ad data = {.data = ad, .len = ad_len};
		
		err = set_ad(adv, &data 1);
```

which is very similar to what you propose, but with the change that we don't accidentally use the same variable for `ad` and `sd`. 

Not sure why `d[2]` was even of size 2, since it's clearly only using `[0]`. 

Could also take it a step further and modify `set_ad` and `set_sd` to take a `struct bt_data` and then handle the `bt_data`=>`bt_ad` conversion at a later point.


In any case, this PR is still a small improvement, so I'm OK accepting it